Третья часть по лабораторной работе Library (события)

PopovMatvey opened the pull request:
По лабораторной работе с событиями был реализован следующий функционал:

1. Добавление новой книги с использованием модального окна;
2. Был реализован поиск по названию книги и по автору;
3. Есть возможность изменить рейтинг при нажатии на название книги
4. Был реализован фильтр "Most Popular".

PopovMatvey committed:
Add files via upload

PopovMatvey committed:
Add files via upload

djilua1 commented on src/library_task/assets/js/eventsHandler.js:
```diff
@@ -38,11 +52,9 @@ addNewBookInput.onclick = function () { // eslint-disable-line func-names
-        ratingBook.value);// eslint-disable-line no-undef
     };
 
     alert('Book has been added');// eslint-disable-line no-alert
```
Удаляй дизейбли линтов, так очень плохо выглядит.

djilua1 commented on src/library_task/assets/js/eventsHandler.js:
```diff
@@ -14,6 +14,16 @@ const authorBook = document.querySelector('#authorBook');
 const ratingBook = document.querySelector('#ratingBook');
 const uploadImageBook = document.querySelector('#uploadImage');
 
+function addNewBook(array, imageHref, // eslint-disable-line no-undef
```
Не удалось добавить книгу.

djilua1 commented on src/library_task/assets/js/script.js:
```diff
@@ -273,7 +277,6 @@ updateNewBook.onclick = function () { // eslint-disable-line func-names
-    console.log(arrayBooks);
     fillAllBooks(arrayBooks, bookListImageBlock);
     modalWindowOnChange.style.display = 'none';
   }
```
Поиск работает не совсем корректно.

djilua1 commented on src/library_task/assets/js/script.js:
```diff
@@ -188,6 +188,10 @@ function createBook(imageHref, titleText, spanText, assessment, parentElement, a
   parentElement.appendChild(bookItemBlock);
```
Поиск книги работает странным образом, появляются книги, которые не удовлетворяют условию.

PopovMatvey committed:
Add files via upload

PopovMatvey committed:
Delete eventsHandler.js

PopovMatvey committed:
Исправления по третей части лаб. работы с Library

Были исправлены все замечания:
1. Удалил все лишние дисейблы линтов, для более удобного отображения;
2. Подправил, поиск книг по автору и названию, теперь поиск происходит только по первому вхождению, строки а не любому.
3. ScrollBar у body был удалён.
4. Добавление книги происходит успешно.

PopovMatvey committed:
Add files via upload

PopovMatvey committed:
Delete script.js

PopovMatvey committed:
Add files via upload
